library/src/main/java/com/getstream/sdk/chat/viewmodel/ChannelViewModelBinding.kt

```diff
@@ -0,0 +1,40 @@
+package com.getstream.sdk.chat.viewmodel
+
+import android.content.Context
+import android.text.format.DateUtils
+import androidx.lifecycle.LifecycleOwner
+import androidx.lifecycle.Observer
+import com.getstream.sdk.chat.R
+import com.getstream.sdk.chat.utils.LlcMigrationUtils
+import com.getstream.sdk.chat.view.ChannelHeaderView
+import io.getstream.chat.android.client.models.Member
+import java.util.isInLastMinute
+
+fun ChannelViewModel.bindView(view: ChannelHeaderView, lifecycleOwner: LifecycleOwner) {
+	members.observe(lifecycleOwner, Observer { members ->
+		view.setHeaderLastActive(members.lastActive(view.context))
+		view.configHeaderAvatar(members)
+	})
+	channelState.observe(lifecycleOwner, Observer { channel ->
+		view.currentChannel = channel
+		LlcMigrationUtils.getChannelNameOrMembers(channel)
+				.takeUnless { it.isBlank() }
+				?.let { view.setHeaderTitle(it) }
+	})
+	anyOtherUsersOnline.observe(lifecycleOwner, Observer { view.setActiveBadge(it) })
+	view.onBackClick = {
+		if (isThread) {
+			resetThread()
+		} else {
+			view.closeActivity()
```

Comment:
Imo, the logic of navigation should be extracted outside the view. For example, in the new sample app there is `androidx.navigation` used to navigate between Fragments. So, I'd like the view model to emit state `State.NavigateBack` and handle it in the fragment.

Eventually, ChannelViewModel will be rewritten in kotlin, so we can refactor this later on.

Reply:
Agreed. Also, new UI from Josh focuses more on replies than threads. (It's a more popular pattern on mobile)

Reply:
I agree. It was migrated on the same way it was previously implemented, to keep functionalities, but I have been thinking about to remove it and try to extract the other "icon actions" out of the view/ViewModel as well

Reply:
On the other hand, I think `ChannelViewModel` should be split, it has lot of logic that could be simplified and only to use what we really need for the View it renders